yamllint/format.py

```diff
@@ -0,0 +1,429 @@
+
+from __future__ import print_function
+
+import os
+import platform
+import sys
+import json
+import datetime
+
+from yamllint.linter import PROBLEM_LEVELS
+
+
+CODECLIMATE_SEVERITY = {
+    None: "info",
+    'warning': "minor",
+    'error': "major",
+}
+
+
+def supports_color():
+    supported_platform = not (platform.system() == 'Windows' and not
+                              ('ANSICON' in os.environ or
+                               ('TERM' in os.environ and
+                                os.environ['TERM'] == 'ANSI')))
+    return (supported_platform and
+            hasattr(sys.stdout, 'isatty') and sys.stdout.isatty())
+
+
+def run_on_gh():
+    """Return if the currnet job is on github."""
+    return 'GITHUB_ACTIONS' in os.environ and 'GITHUB_WORKFLOW' in os.environ
+
+
+def escape_xml(text):
+    """Escape text for XML."""
+    text = text.replace('&', '&amp;')
+    text = text.replace('<', '&lt;')
+    text = text.replace('>', '&gt;')
+    text = text.replace('"', '&quot;')
+    text = text.replace("'", '&apos;')
+    return text
+
+
+def severity_from_level(level):
+    if isinstance(level, int):
+        level = PROBLEM_LEVELS[level]
+    return CODECLIMATE_SEVERITY[level]
+
+
+class Formater(object):
+    """Any formater."""
+    # the formater name
+    name = ''
+
+    @classmethod
+    def get_formaters_names(cls):
+        """Return all formaters names."""
+        return [f.name for f in cls.__subclasses__()]
+
+    @classmethod
+    def get_formater(cls, name, no_warn):
+        """Return a formater instance."""
+
+        if name == 'auto':
+            if run_on_gh():
+                name = 'github'
+            elif supports_color():
+                name = 'colored'
+            else:
+                name = 'standard'
```

Comment:
This is the uncovered code (with also `run_on_gh` & `supports_color`).

I don't know how to test it because it's platform dependent...